src/docs/frontend/component-library.mdx

```diff
@@ -0,0 +1,151 @@
+});
+```
+
+This template includes all the basics:
+
+1. The filename is `button.stories.tsx` and sits directly beside `button.tsx` which is the component being described.
+2. The file uses a default export, there is no need to name things when there is only one export.
+3. To create the `storyBook()` we're passing in the `Button` component directly as the first argument. This sets the title at the top of the page, and avoids typos.
+4. We've implemented a function that accepts `story` as it's argument. This is inspired by `jest` and the [`describe()`](https://jestjs.io/docs/api#describename-fn) function.
+5. Each time we call `story()` we can render out a unit onto the page. This includes a title, and the rendered component example. This is similar to `jest` and the [`test()` or `it()`](https://jestjs.io/docs/api#testname-fn-timeout) function, with the exception that `story` is not a global.
+    
+Using this framework you get some nice things automatically:
+
+- Stories are printed in the same order they are written in the source file
+- Each `story()` callback can be async
+
+Some tips to remember:
+
+- Don't be afraid to import anything else you need to help illustrate use cases for your component. It's useful to leverage `<p>` and `<Fragment>` in most cases for example.
+- If you are demonstrating a component that relies on a global hook like `useOrganization()` or `usePageFilters()`, then you should wrap your story with those context providers manually.
+- Named exports, or a mix of named and default exports are all supported. This is more useful with [non-framework stories](#non-framework-stories).
+- If you are testing a hook then you must pass a string as the first argument to `storyBook()` instead of the hook itself because.
+
+### Non-framework stories
+
+It is possible to skip the `storyBook()` framework function and build stories from scratch. 
+
+Doing this allows full flexibility to define your story.
+
+Using default exports, named exports, or a mixture of each is fully supported. But take note that the order they will be rendered is not consistent/defined.
+
+Look at [icons.stories.tsx](https://sentry.io/orgredirect/organizations/:orgslug/stories/?name=app/icons/icons.stories.tsx) for an example.
+
+### Helper Components
+
+There are some helper components specifically built to make common patterns in stories easier to implement. Of course you can import anything in the sentry repo to make your stories richer, and create new helpers to make expressing stories easier!
+
+Helper Components:
+
+<IconFire color="gray500" size="sm" />
+
+- [`<JSXNode />`](https://github.com/getsentry/sentry/blob/master/static/app/components/stories/jsxNode.tsx)<br />
+    Render a formatted JSX component name and some properties.<br />
+    ```typescript
+    return <JSXNode name="IconFire" props={{color: 'red400', size: 'sm'}} />;
+
+    // will render as <code>&lt;IconFire<br/>color="red"<br/>size="sm" \:gt;</code>
```

Comment:
are the `&lt;` and `\:gt;` intentional?

Reply:
Yes. Trying to illustrate that we will output visible angle-braces to the screen. `&gt;` is the html escape code for that.